guidance/models/transformers/_transformers.py

```diff
@@ -174,6 +76,203 @@ def __init__(
             transformers_tokenizer.eos_token_id,
         )
 
+    def _tokenizer(self, model: str, **kwargs) -> Union[
+        "transformers_package.PreTrainedTokenizer",
+        "transformers_package.PreTrainedTokenizerFast",
+    ]:
+        # make sure transformers is installed
+        if not has_transformers:
+            raise Exception("Please install transformers with `pip install transformers`")
+
+        try:
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=False, **kwargs
+            )
+        # except ImportError:
+        #     # HuggingFace needs us to install something (sentencepiece, protobuf, etc. for some non-fast tokenizers)
+        #     # TODO: decide whether to warn and fallback to fast tokenizer or to raise (should at least warn...)
+        #     raise
```

Comment:
Uncommenting this would cause huggingface to raise an exception about sentencepiece early. With it commented out, we'll try a few approaches to getting the byte_tokens and then eventually raise our own exception if we couldn't make it work.